docs/development-api/github.md

```diff
@@ -20,6 +20,56 @@ const buildName = GitHub.buildNameFromEnvironment();
 
 ---
 
+## pushFileChanges
+
+Pushes file changes from the local workspace to a specified GitHub branch.
+
+These file changes will appear as verified commits on GitHub.
+
+Please note: this will not update the local Git repository
+
+```typescript
+import { GitHub } from 'skuba';
+
+const commitId = await GitHub.pushFileChanges({
+  dir: './',
+  branch: 'some-branch',
+  fileChanges: {
+    additions: [{ contents: '', path: 'another-path' }],
+    deletions: [{ path: 'some-path' }],
+  },
+  messageHeadline: 'some-commit',
+  messageBody: 'extra-body',
+});
+```
+
+---
+
+## pushAllFileChanges
+
+Retrieves all file changes from the local Git repository using [getChangedFiles],
+and then pushes the changes to a specified GitHub branch using [pushFileChanges](#pushFileChanges).
+
+These file changes will appear as verified commits on GitHub.
+
+`undefined` is returned instead of a commit ID when there are no changes to push.
+
+Please note: this will not update the local Git repository by default unless `updateLocal` is specified.
```

Comment:
```suggestion
This function is roughly equivalent to `git add --all && git commit && git push`,
but it will not update the local Git repository unless `updateLocal` is specified.
```